### PR TITLE
Reusable role button component

### PR DESCRIPTION
**Pull request by pzduniak**
image
image
image

### CODE REVIEW

File: shared/common-adapters/dropdown.tsx [hunk 1]
```diff
@@ -112,14 +112,15 @@ class Dropdown<N extends React.ReactNode> extends React.Component<Props<N> & Ove
...
+  label: React.ReactNode
   onPress: () => void
-  type: 'Body' | 'BodySmall'
+  type: 'Body' | 'BodySmall' | 'BodySmallSemibold'
+  style?: Styles.StylesCrossPlatform
 }
 
 export const InlineDropdown = (props: InlineDropdownProps) => {
   const selected = (
-    <Box2 direction="horizontal" key={props.label} style={styles.inlineSelected}>
```
Comment (jakob223) [inline shared/common-adapters/dropdown.tsx]: presumably this had a key for a reason.. are you sure you can just remove it?

File: shared/teams/role-button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+const RoleButton = (props: Props) => {
+  const iconType = roleIconMap[props.selectedRole]
+
+  return (
+    <InlineDropdown
+      type="BodySmallSemibold"
+      label={
+        <Kb.Box2 direction="horizontal" alignItems="center" style={styles.label}>
+          <Kb.Icon type={iconType} style={styles.icon} fontSize={Styles.isMobile ? 16 : 12} />
```
Comment (jakob223) [inline shared/teams/role-button/index.tsx]: I think this fontSize business is equivalent to `sizeType="Default"` which is preferable.

File: shared/common-adapters/dropdown.tsx (hunk 1 above)
Comment (pzduniak) [reply shared/common-adapters/dropdown.tsx]: Checked, it is not and was not needed.

File: shared/teams/role-button/index.tsx (hunk 2 above)
Comment (pzduniak) [reply shared/teams/role-button/index.tsx]: It's in fact `Small`, I forgot about `sizeType`, cheers!

**Pull request merged**